tests/misc_test.go

```diff
@@ -635,3 +635,32 @@ func TestTypeConversion(t *testing.T) {
 		t.Fail()
 	}
 }
+
+// See https://github.com/gopherjs/gopherjs/issues/851.
+func TestSlicingNilSlice(t *testing.T) {
+	t.Run("StaysNil", func(t *testing.T) {
+		var s []int
+		s = s[:]
+		s = s[0:0]
+		s = s[0:0:0]
+		if s != nil {
+			t.Errorf("nil slice became non-nil after slicing: %#v", s)
```

Comment:
I know the expectation, but would it be better to reword this including expectation?